### PR TITLE
Add various comments as FIXMEs

### PR DESCRIPTION
**Pull request by schanzen**
Related to #7 and #8

Did a pass of the document and was wondering to what degree it is supposed to be normative. If it is **not** supposed to be, it should be made clear(er) what the diff over the respective papers (informative references to those missing) is.
Also, normative references (such as those to the HKDF) are not required then.

If it is supposed to be normative, it requires the definition of (at least) one concrete instantiation (with hash functions/curves) and normative test vectors and wire formats.

Once we can clarify those points I can replace the FIXMEs aligning with the target.

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -836,6 +842,8 @@ The following comparison assumes BBS signatures with curve BLS12-381, targeting
 
 For 128 bits security, ECDSA with curve P-256 takes 37 and 79 micro-seconds to sign and verify signature on a modern computer. BBS 680 and 1400 milliseconds to sign and verify a single message. However, ECDSA can only sign a single message whereas BBS can sign any number of messages at the expense of a bigger public key. To sign and verify 10 messages takes 3.7 and 5.4 milliseconds, and 22.3 and 24.4 milliseconds for 100 messages.
 
+(FIXME: How is sign and verify of 10 messages faster than a single message? Also, it seems to scale linearly, as would an ECDSA signature over 10/100 messages)
```
Comment (andrewwhitehead) [inline spec.md:845]: I think those 680 and 1400 numbers are also in microseconds not milliseconds?

File: spec.md [hunk 2]
```diff
@@ -52,20 +52,19 @@ A digital signature scheme is a fundamental cryptographic primitive that is used
...
 
-This document describes the BBS signature scheme. The scheme features many important properties:
+This document describes the BBS signature scheme. (FIXME: THE BBS signature scheme? Or A BBS signature schemes? Also: BBS citation needed/informative reference).
+The scheme feature important properties that allow the scheme to be used in applications where privacy and data minimization techniques are desired and/or required:
 
-1. The signature is over a group of Pedersen commitments--signatures can be created blinded or un-blinded.
+1. Signatures can be created blinded or un-blinded.
    
-2. The signature is encoded as a single group element and two field elements.
+2. Signatures are encoded as a single group element and two field elements. (FIXME: Why is this important? Succinctness? Unclear)
```
Comment (andrewwhitehead) [inline spec.md:60]: Why does it need justification?

File: spec.md [hunk 3]
```diff
@@ -52,20 +52,19 @@ A digital signature scheme is a fundamental cryptographic primitive that is used
...
 
-3. Verification requires 2 pairing operations.
+3. Verification requires 2 pairing operations. (FIXME: Only two? Why is this important? It's not as good as one or no pairing!)
```
Comment (andrewwhitehead) [inline spec.md:62]: Same here, not sure what the suggested change would be.

File: spec.md (hunk 2 above)
Comment (schanzen) [reply spec.md:60]: Because above it it says "The scheme features many important properties:"
It is unclear why this generic property is important.
I changed this sentence above in the pull request already.
If this is just a general note then it is not an important property of the scheme I think.

File: spec.md (hunk 1 above)
Comment (schanzen) [reply spec.md:845]: It also kind of implies that there is space time tradeoff and that BBS can sign any number of messages in 680/1400ms. So maybe the 3.7 and 5.4 actually refer to ECDSA. Not sure.

File: spec.md (hunk 3 above)
Comment (schanzen) [reply spec.md:62]: Well let me rephrase it using the words from the (original) text:
How is "Verification requires 2 pairing operations." a "[property that] allow[s] the scheme to be used in applications where privacy and data minimization techniques are desired and/or required."?
That is what the text currently says, and it is unclear (to me) where the connection is.

File: spec.md (hunk 2 above)
Comment (schanzen) [reply spec.md:60]: Also, the "encoding" of the signature would usually refer to a normative (interoperable) wire format of the signature. But the encoding  Maybe what is meant here is that the signatures are "represented" as a single group element and two field elements?
Later in the text it is also stated that "How a signature is to be encoded is not covered by this document. (TODO perhaps add some additional information in the appendix)"

File: spec.md [hunk 4]
```diff
@@ -690,6 +695,7 @@ Procedure:
 28. return spk
 
 How a signature is to be encoded is not covered by this document. (TODO perhaps add some additional information in the appendix)
+(FIXME: Encoding out of scope OK, but wire format is required for test vectors. Unless this document is not normative.)
```
Comment (tplooker) [inline spec.md:698]: I think I agree with your comment here but could you clarify what you mean by "wire format" specifically and what you would take as the definition of encoding format?

File: spec.md [hunk 5]
```diff
@@ -262,6 +264,8 @@ Procedure:
 SkToDpk algorithm takes a secret key SK and outputs a corresponding short formed public key.
 
 SK MUST be indistinguishable from uniformly random modulo r (Section 2.2) and infeasible to guess, e.g., generated using a trusted source of randomness.  KeyGen (Section 2.3) outputs SK meeting these requirements.  Other key generation approaches meeting these requirements MAY also be used; details of such methods are beyond the scope of this document.
+(FIXME: This paragraph is a duplicate from Section 2.2. where it is already established that IKM is random and cannot be guessed)
```
Comment (tplooker) [inline spec.md:267]: If it is a duplicate then I think we should just remove it

File: spec.md (hunk 1 above)
Comment (mikelodder7) [reply spec.md:845]: Sure feel free to remove it.

File: spec.md (hunk 4 above)
Comment (schanzen) [reply spec.md:698]: As an implementer I would expect qualifiers such as "32 octets" or "variable length" (with a length field) for the members of the tuple and then a normative ordering of them (for example by having a wire format graphic or a concatenation of the octet strings).
The point is that when I eventually code my structs, I need to know the size of its members and their location.
Otherwise I do not even know how to parse a test vector (if provided).
This unclarity may have to do with #8 which makes normative statements more difficult to incorporate.

File: spec.md (hunk 2 above)
Comment (tplooker) [reply spec.md:60]: Yeah I think this is covered by issue #2 and we should aim to review it on the next working group call to provide better clarity around the direction the WG would like to take this

File: spec.md (hunk 4 above)
Comment (tplooker) [reply spec.md:698]: Yeah I think this is covered by issue #2 and we should aim to review it on the next working group call to provide better clarity around the direction the WG would like to take this

File: spec.md (hunk 1 above)
Comment (tplooker) [reply spec.md:845]: @schanzen, could you update your PR to just remove this section entirely from the appendix?

File: spec.md (hunk 5 above)
Comment (tplooker) [reply spec.md:267]: Have reviewed, @schanzen can you remove the duplicate language around SK, then we can resolve this FIX me

File: spec.md (hunk 2 above)
Comment (tplooker) [reply spec.md:60]: +1 reading the context it does not appear apparent why this needs to be bulleted as it is, I would vote to remove it.

File: spec.md (hunk 3 above)
Comment (tplooker) [reply spec.md:62]: +1 as above I think this should be removed also

**Review by andrewwhitehead — APPROVED** [review]

**Review by tplooker — APPROVED** [review]

**Comment by tplooker** [comment]
Purely editorial changes, multiple approvals, discussed on WG, merging, thanks @schanzen
